File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group_anagram = {}
        for word in strs:
            group_anagram.setdefault(''.join(sorted(word)), []).append(word)
        
        return list(group_anagram.values())

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hm = new HashMap<>();

        for (String s: strs) {
            char[] c = s.toCharArray();
            Arrays.sort(c);
            String t = new String(c);

            hm.computeIfAbsent(t, k -> new ArrayList<>()).add(s);
        }

        List<List<String>> res = new ArrayList<>();

        for (List<String> ls : hm.values()) {
            res.add(ls);
        }

        return res;
    }
}